modules/account-lib/src/coin/algo/transactionBuilder.ts

```diff
@@ -1,44 +1,475 @@
-import BigNumber from 'bignumber.js';
 import { BaseCoin as CoinConfig } from '@bitgo/statics';
-import { BaseTransactionBuilder } from '../baseCoin';
-import { NotImplementedError } from '../baseCoin/errors';
-import { BaseAddress, BaseKey } from '../baseCoin/iface';
+import BigNumber from 'bignumber.js';
+import algosdk from 'algosdk';
+import { BaseTransactionBuilder, TransactionType } from '../baseCoin';
+import { BuildTransactionError, InvalidTransactionError } from '../baseCoin/errors';
+import { BaseAddress, BaseFee, BaseKey } from '../baseCoin/iface';
+import { isValidEd25519Seed } from '../../utils/crypto';
 import { Transaction } from './transaction';
+import { AddressValidationError, InsufficientFeeError } from './errors';
+import { KeyPair } from './keyPair';
+import { BaseTransactionSchema } from './txnSchema';
+import Utils from './utils';
+
+const MIN_FEE = 1000; // in microalgos
+
+const MAINNET_GENESIS_ID = 'mainnet-v1.0';
+const MAINNET_GENESIS_HASH = 'wGHE2Pwdvd7S12BL5FaOP20EGYesN73ktiC1qzkkit8=';
+const TESTNET_GENESIS_ID = 'testnet-v1.0';
+const TESTNET_GENESIS_HASH = 'SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=';
+const BETANET_GENESIS_ID = 'betanet-v1.0';
+const BETANET_GENESIS_HASH = 'mFgazF+2uRS1tMiL9dsj01hJGySEmPN28B/TjjvpVW0=';
 
 export abstract class TransactionBuilder extends BaseTransactionBuilder {
-  private _transaction: Transaction;
+  protected _transaction: Transaction;
+  protected _keyPairs: KeyPair[];
+
+  // the fee is specified as a number here instead of a big number because
+  // the algosdk also specifies it as a number.
+  protected _fee: number;
+  protected _isFlatFee: boolean;
+
+  protected _sender: string;
+  protected _genesisHash: string;
+  protected _genesisId: string;
+  protected _firstRound: number;
+  protected _lastRound: number;
+  protected _lease?: Uint8Array;
+  protected _note?: Uint8Array;
+  protected _reKeyTo?: string;
+  protected _suggestedParams: algosdk.SuggestedParams;
 
   constructor(coinConfig: Readonly<CoinConfig>) {
     super(coinConfig);
+
+    this._transaction = new Transaction(coinConfig);
+    this._keyPairs = [];
+  }
+
+  /**
+   * Sets the fee.
+   *
+   * The minimum fee is 1000 microalgos.
+   *
+   * @param {BaseFee} feeObj The amount to pay to the fee sink denoted in microalgos
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://developer.algorand.org/docs/reference/transactions/
+   */
+  fee(feeObj: BaseFee): this {
+    this._isFlatFee = true;
+    const fee = new BigNumber(feeObj.fee).toNumber();
```

Comment:
BigInt is a string. We need a number here.

Reply:
I don't quite follow what you mean by "BigInt is a string" - `bigint` and `string` are distinct types. I think this is safe since we can assume that the fee amount is representable by a normal js Number.